features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/DeviceConfigServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+import org.opennms.netmgt.poller.support.SimpleMonitoredService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class DeviceConfigServiceImpl implements DeviceConfigService {
+
+    private static final String DEVICE_CONFIG_SERVICE_NAME_PREFIX = "DeviceConfig-";
+    private static final String DEVICE_CONFIG_MONITOR_CLASS_NAME = "org.opennms.features.deviceconfig.monitors.DeviceConfigMonitor";
+    private static final Logger LOG = LoggerFactory.getLogger(DeviceConfigServiceImpl.class);
+    @Autowired
+    private LocationAwarePollerClient locationAwarePollerClient;
+
+    @Autowired
+    private SessionUtils sessionUtils;
+
+    @Autowired
+    private IpInterfaceDao ipInterfaceDao;
+
+    @Qualifier("deviceConfigMonitorAdaptor")
+    private ServiceMonitorAdaptor serviceMonitorAdaptor;
+
+
+    @Override
+    public void triggerConfigBackup(String ipAddress, String location, String configType) throws IOException {
+
+        if (Strings.isNullOrEmpty(configType)) {
+            configType = ConfigType.Default.name();
+        }
+        String serviceName = DEVICE_CONFIG_SERVICE_NAME_PREFIX + configType;
+        MonitoredService service = sessionUtils.withReadOnlyTransaction(() -> {
+
+            Optional<OnmsIpInterface> ipInterfaceOptional = ipInterfaceDao.findByIpAddress(ipAddress).stream().filter(ipInterface ->
+                    ipInterface.getNode().getLocation().getLocationName().equals(location)).findFirst();
+            if (ipInterfaceOptional.isEmpty()) {
+                return null;
+            }
+            OnmsIpInterface ipInterface = ipInterfaceOptional.get();
+            OnmsNode node = ipInterface.getNode();
+
+            return new SimpleMonitoredService(ipInterface.getIpAddress(), node.getId(), node.getLabel(), serviceName, location);
+        });
+
+        if (service == null) {
+            throw new IllegalArgumentException("No interface found with ipAddress " + ipAddress + " at location " + location);
+        }
+        // All the service parameters should be loaded from metadata in PollerRequestBuilderImpl
+        // Persistence will be performed in DeviceConfigMonitorAdaptor.
+        final CompletableFuture<PollerResponse> future = locationAwarePollerClient.poll()
+                .withService(service)
+                .withAdaptor(serviceMonitorAdaptor)
+                .withMonitorClassName(DEVICE_CONFIG_MONITOR_CLASS_NAME)
+                .execute();
+        future.whenComplete(((pollerResponse, throwable) -> {
+            if (throwable != null) {
+                LOG.info("Error while manually triggering config backup for IpAddress {} at location {} for service {}", ipAddress, location, service);
```

Comment:
Maybe we want the exception to be also logged? Just add it as an extra argument.

```suggestion
                LOG.info("Error while manually triggering config backup for IpAddress {} at location {} for service {}", ipAddress, location, service, throwable);
```